ResourceStatusSystem/Agent/EmailAgent.py

```diff
@@ -38,55 +41,25 @@ def initialize( self ):
 
   def execute( self ):
 
-    #if the file exists and it is not empty
-    if os.path.isfile(self.cacheFile) and os.stat(self.cacheFile).st_size:
+    conn = sqlite3.connect(self.cacheFile)
 
-      #rename the file and work with it in order to avoid race condition
-      os.rename(self.cacheFile, self.cacheFile + ".send")
-      self.cacheFile += ".send"
+    result = conn.execute("SELECT DISTINCT SiteName from ResourceStatusCache;")
+    for site in result:
+      cursor = conn.execute("SELECT StatusType, ResourceName, Status, Time, PreviousStatus from ResourceStatusCache WHERE SiteName='"+ site[0] +"';")
 
-      #load the file
-      with open(self.cacheFile, 'r') as f:
-        new_dict = json.load(f)
+      email_body = ""
+      for row in cursor:
```

Comment:
it is maybe more readable to unpack the values. For example 
``` for x, y, z in cursor```

This way, we know what is the meaning of ```row[0], row[1]...```